hadoop-hdds/container-service/src/main/java/org/apache/hadoop/ozone/container/common/transport/server/ratis/ContainerStateMachine.java

```diff
@@ -862,4 +864,48 @@ public void notifyLeaderChanged(RaftGroupMemberId groupMemberId,
                                   RaftPeerId raftPeerId) {
     ratisServer.handleLeaderChangedNotification(groupMemberId, raftPeerId);
   }
+
+  @Override
+  public String toStateMachineLogEntryString(StateMachineLogEntryProto proto) {
+    try {
+      ContainerCommandRequestProto requestProto =
+              getContainerCommandRequestProto(proto.getLogData());
+      return getWriteChunkInfo(requestProto);
+    } catch (Throwable t) {
+      return "";
+    }
+  }
+
+  private String getWriteChunkInfo(ContainerCommandRequestProto requestProto) {
+    StringBuilder builder = new StringBuilder();
+    Preconditions.checkArgument(requestProto.getCmdType() == Type.WriteChunk);
+
+    long contId = requestProto.getContainerID();
+    WriteChunkRequestProto wc = requestProto.getWriteChunk();
+
+    builder.append("cmd=");
+    builder.append(requestProto.getCmdType().toString());
+
+    builder.append(", container id=");
+    builder.append(contId);
+
+    builder.append(", blockid=");
+    builder.append(contId);
+    builder.append(":localid=");
+    builder.append(wc.getBlockID().getLocalID());
+
+    builder.append(", chunk=");
+    builder.append(wc.getChunkData().getChunkName());
+    builder.append(":offset=");
+    builder.append(wc.getChunkData().getOffset());
+    builder.append(":length=");
+    builder.append(wc.getChunkData().getLen());
+
+    Container cont = containerController.getContainer(contId);
+    if (cont != null) {
+      builder.append(", container path=");
+      builder.append(cont.getContainerData().getContainerPath());
+    }
+    return builder.toString();
```

Comment:
Can we add the string creation logic in a utils class?

Reply:
Done.